server/api/controllers/image-controller.ts

```diff
@@ -1,215 +1,219 @@
+    try {
+      const imageFiles = Array.isArray(req.files?.images)
+        ? (req.files?.images as UploadedFile[])
+        : [req.files?.images as UploadedFile];
+
+      if (!imageFiles.length) {
+        throw new Error("No images were uploaded");
+      }
+
+      const errors: string[] = [];
+      const imagesWithSupportedFileData: ImageWithFile[] = [];
+
+      for (const imageFile of imageFiles) {
+        if (!imageFile.mimetype) {
+          errors.push(
+            `No mimetype/filetype provided with file ${imageFile.name}`
+          );
+          continue;
+        }
+
+        if (imageFile.mimetype.match(/(^image)(\/)[a-zA-Z0-9_]*/gm) === null) {
+          errors.push(
+            `Unsupported filetype for ${imageFile.name}. We only support image file types.`
+          );
+          continue;
+        }
+
+        const transformedImage: ImageWithFile = {
+          fileName: imageFile.name,
+          fileType: imageFile.name.split(".").slice(1).join("."),
+          path: this.getFilePath(imageFile.name),
+          publicLink: this.getPublicLink(imageFile.name),
+          file: imageFile,
+        };
+
+        imagesWithSupportedFileData.push(transformedImage);
+      }
+
+      this.uploadQueue.push({
+        images: imagesWithSupportedFileData,
+        errors,
+        res,
+      });
+    } catch (err) {
+      this.logger.error(err.message);
+      res.status(400).redirect("/admin/image-store");
+      return;
+    }
+  }
+
+  /*
+  Handles the /api/image/delete endpoint
+  - checks that file was passed
+  - checks that file exists
+  - if validations pass, adds the delete request to the ImageRequestQueue
+  */
+  deleteFile(req: Request, res: Response): void {
+    try {
+      const { fileName, fileType, path, publicLink } = req.body;
+
+      if (!path) {
+        throw new Error("File path not provided with delete request");
+      }
+
+      if (!existsSync(path)) {
+        throw new Error("Faulty file path provided with delete request");
+      }
+
+      const transformedImage: Image = {
+        fileName,
+        fileType,
+        path,
+        publicLink,
+      };
+
+      this.deleteQueue.push({ ...transformedImage, res });
+    } catch (err) {
+      this.logger.error(err.message);
+      res.status(400).redirect("/admin/image-store");
+      return;
+    }
+  }
+
+  async processFileUpload(request: ImageUploadRequest): Promise<void> {
+    const { images, errors, res } = request;
+
+    try {
+      for (const image of images) {
+        await image.file?.mv(image.path);
+        this.logger.info(`${image.fileName} uploaded to ${image.path}`);
+      }
+
+      this.rewriteFileJson();
+    } catch (err) {
+      this.logger.error(err);
+    }
+
+    const urlSearchParams = new URLSearchParams();
+    errors.forEach((item) => urlSearchParams.append("errors", item));
+    res.status(201).send({
+      status: "success",
+      errors: errors,
+      files: images.map((im) => im.fileName),
+    });

```

Comment:
This is the only change in this file - we return the list of filenames uploaded to the server.